### PR TITLE
SizeBay Challenge Finished

### PR DESCRIPTION
**Pull request by ThiagoArndt**

### CODE REVIEW

File: src/components/add/AddItem.tsx
Comment (notlesz) [inline src/components/add/AddItem.tsx:1]: You don't need to create a state to control when the icon should be disabled or not. You can just control it with the state `userInputValue`.

```
const isIconActive = !!userInputValue.length
```

File: src/components/items/Item.tsx
Comment (notlesz) [inline src/components/items/Item.tsx:1]: The situation is the same as before. You're using a state to control when the task is completed but you already get the isDone property in the props.
Another point: you've added a reference to the input, but you're not actually using it.

File: src/components/layout/MainHeader.tsx
Comment (notlesz) [inline src/components/layout/MainHeader.tsx:1]: Nice, you've chosen to format the date natively! But how about improving the code? That way you have the properties ready without having to format them one by one

```
const [weekday, monthDay, year] = date.toLocaleDateString('en-US', {
  weekday: 'long',
  year: 'numeric',
  month: 'short',
  day: 'numeric',
}).split(", ");

const [month, day] = monthDay.split(" ")

```

File: src/components/search/FilterSection.tsx
Comment (notlesz) [inline src/components/search/FilterSection.tsx:1]: This component has an unnecessary responsibility. You can choose to create a `filter` property (for example) on the redux and control which filter is active. The same goes for search.
Leave the responsibility of filtering the tasks to the moment you render them.

File: src/utils/enums.ts
Comment (notlesz) [inline src/utils/enums.ts:1]: You could replace "Null" to "All", since it represents when you are listing all the tasks and thus improving the DX

**Review by notlesz — CHANGES_REQUESTED** [review]
Hi [@ThiagoArndt](https://github.com/ThiagoArndt)!, whats up? Here's a review of your technical test for the frontend position. More detailed points related to your code will be commented on directly.

**One tip:** it would be good if you deployed the application on Vercel or on some other deployment platform so that we could have easy access to the application.

# Layout

I've noticed a few things that can be improved and will increase the quality of your challenge submission:

1.  Progress bar init completed without items registered.
2.  Use semantic HTML tags, e.g., main, section, so on so forth
3.  Font family, font size and colors don't match the design.
4.  The design components don't match the design.
5.  Add the submit event on item creation input to improve user experience.
6.  Add warning text when you can't find items when searching or filtering.
7.  Data is not persisting in localStorage.

# Code

1.  Out of curiosity: why did you choose to use NextJS for the challenge, instead of going with something simpler like Vite + React Router? You're not using its features so it becomes unnecessary complexity in the end, with a large bundle size and room for performance issues.
2.  Make sure to use eslint, prettier or both to improve the formatting of your code.
3.  There are a lot of unnecessary complexity to some of the logic used to solve your problems.
4.  You did not used any of the lib-recommended tools that React provides (useMemo, useCallback, etc.) that can improve your code and consequently improve performance.
5.  Folder structure of the components can be improved, you did not specified what each one actually is. E.g If your component has a sub-component, create a file for it along with its style, this will greatly improve the structure of the project.

# Final thoughts

After reviewing your frontend developer challenge, we recognize your effort and dedication to completing it. However, we have identified areas for improvement that can bring your work into line with the project's expectations.
It is essential to adhere more faithfully to the design provided, paying attention to details such as fonts, colors, and the construction of its components. In a real scenario this makes all the difference.
The user experience is the main focus when building projects, and you can enhance it with features such as data persistence and warning messages for blank states and white spaces.
Re-evaluate the need for NextJS for this project, in order to simplify and avoid unnecessary complexity.
We recommend that you read the [React documentation](https://react.dev/) and better understand in which scenarios you should use the available features to solve your problems and thus improve performance.

If you have any questions, we'd be happy to help you making these improvements. When you're ready, leave a comment so we can resume the analysis. Thank you!

**Comment by vaporwavie** [comment]
Hi @ThiagoArndt! Do you need any help from me or @notlesz to address the review changes? Let us know if you need any assistance!

**Comment by vaporwavie** [comment]
Closing this due to inactivity. If you're willing to get back to this, open a new PR and we can proceed with reviewing it. Thanks!

**Pull request closed without merging**